test/helm_task_test.go

```diff
@@ -0,0 +1,411 @@
+
+	knativetest.CleanupOnInterrupt(func() { tearDown(logger, c.KubeClient, namespace) }, logger)
+	defer tearDown(logger, c.KubeClient, namespace)
+
+	logger.Infof("Creating Git PipelineResource %s", sourceResourceName)
+	if _, err := c.PipelineResourceClient.Create(getGoHelloworldGitResource(namespace)); err != nil {
+		t.Fatalf("Failed to create Pipeline Resource `%s`: %s", sourceResourceName, err)
+	}
+
+	logger.Infof("Creating Task %s", createImageTaskName)
+	if _, err := c.TaskClient.Create(getCreateImageTask(namespace, t)); err != nil {
+		t.Fatalf("Failed to create Task `%s`: %s", createImageTaskName, err)
+	}
+
+	logger.Infof("Creating Task %s", helmDeployTaskName)
+	if _, err := c.TaskClient.Create(getHelmDeployTask(namespace)); err != nil {
+		t.Fatalf("Failed to create Task `%s`: %s", helmDeployTaskName, err)
+	}
+
+	logger.Infof("Creating Pipeline %s", helmDeployPipelineName)
+	if _, err := c.PipelineClient.Create(getHelmDeployPipeline(namespace)); err != nil {
+		t.Fatalf("Failed to create Pipeline `%s`: %s", helmDeployPipelineName, err)
+	}
+
+	logger.Infof("Creating PipelineRun %s", helmDeployPipelineRunName)
+	if _, err := c.PipelineRunClient.Create(getHelmDeployPipelineRun(namespace)); err != nil {
+		t.Fatalf("Failed to create Pipeline `%s`: %s", helmDeployPipelineRunName, err)
+	}
+
+	// Verify status of PipelineRun (wait for it)
+	if err := WaitForPipelineRunState(c, helmDeployPipelineRunName, func(pr *v1alpha1.PipelineRun) (bool, error) {
+		c := pr.Status.GetCondition(duckv1alpha1.ConditionSucceeded)
+		if c != nil {
+			if c.Status == corev1.ConditionTrue {
+				return true, nil
+			} else if c.Status == corev1.ConditionFalse {
+				return true, fmt.Errorf("pipeline run %s failed!", helmDeployPipelineRunName)
+			}
+		}
+		return false, nil
+	}, "PipelineRunCompleted"); err != nil {
+		t.Errorf("Error waiting for PipelineRun %s to finish: %s", helmDeployPipelineRunName, err)
+	}
+
+	logger.Info("Waiting for service to get external IP")
+	var serviceIp string
+	if err := WaitForServiceExternalIPState(c, namespace, helmDeployServiceName, func(svc *corev1.Service) (bool, error) {
+		ingress := svc.Status.LoadBalancer.Ingress
+		if ingress != nil {
+			if len(ingress) > 0 {
+				serviceIp = ingress[0].IP
+				return true, nil
+			}
+		}
+		return false, nil
+	}, "ServiceExternalIPisReady"); err != nil {
+		t.Errorf("Error waiting for Service %s to get an external IP: %s", helmDeployServiceName, err)
+	}
+
+	if serviceIp != "" {
+		resp, err := http.Get(fmt.Sprintf("http://%s:8080", serviceIp))
+		if err != nil {
+			t.Errorf("Error reaching service at http://%s:8080 %s", serviceIp, err)
+		}
+		if resp != nil && resp.StatusCode != http.StatusOK {
+			t.Errorf("Error from service at http://%s:8080 %s", serviceIp, err)
+		}
+
+	} else {
+		t.Errorf("Service IP is empty.")
+	}
+
+	// cleanup task to remove helm from cluster, will not fail the test if it fails, just log
+	removeHelmFromCluster(c, t, namespace, logger)
```

Comment:
i think we should make sure this runs if the test is interrupted or fails, e.g. like we do with the `teardown` (and/or move it to the teardown):

```
	knativetest.CleanupOnInterrupt(func() { tearDown(logger, c.KubeClient, namespace) }, logger)
	defer tearDown(logger, c.KubeClient, namespace)
```

Reply:
ok